caffe2 | MSVS compatibility fixes

boguscoder opened the pull request:
Summary: Code changes required to build caffe2 for windows with toolchain used by FB.

Differential Revision: D13953258

boguscoder commented:
can anyone with write perm. re-trigger two failed checks, please, they look intermittent

peterjc123 commented:
@pytorchbot retest it please

peterjc123 commented on caffe2/operators/quantized/int8_utils.h:
```diff
@@ -143,7 +143,11 @@ activationLimits(float scale, int32_t zero_point, Activation Ac) {
               std::numeric_limits<uint8_t>::max()};
     default:
+#ifdef _MSC_VER
+      break;
```
The equivalent should be __assume(0).

peterjc123 commented:
@pytorchbot retest this please

Commit:
caffe2 | MSVS compatibility fixes (#16765)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16765

Code changes required to build caffe2 for windows with toolchain used by FB.

Differential Revision: D13953258

fbshipit-source-id: e3eee4b430bc58320dd454f7299741db88680a31

boguscoder commented:
@peterjc123 where shall I seek help with tests failed, there are only few of them but they dont seem to be related to my changes
